homeassistant/components/monzo/sensor.py

```diff
@@ -0,0 +1,130 @@
+class MonzoSensorEntityDescriptionMixin:
+    """Adds fields for Monzo sensors."""
+
+    value: Callable[[dict[str, Any]], Any]
+
+
+@dataclass
+class MonzoSensorEntityDescription(
+    SensorEntityDescription, MonzoSensorEntityDescriptionMixin
+):
+    """Describes Monzo sensor entity."""
+
+
+ACC_SENSORS = (
+    MonzoSensorEntityDescription(
+        key="balance",
+        name="Balance",
+        value=lambda data: data["balance"]["balance"] / 100,
+        device_class=SensorDeviceClass.MONETARY,
+        native_unit_of_measurement="£",
+        suggested_display_precision=2,
+    ),
+    MonzoSensorEntityDescription(
+        key="total_balance",
+        name="Total Balance",
+        value=lambda data: data["balance"]["total_balance"] / 100,
+        device_class=SensorDeviceClass.MONETARY,
+        native_unit_of_measurement="£",
+        suggested_display_precision=2,
+    ),
+)
+
+POT_SENSORS = (
+    MonzoSensorEntityDescription(
+        key="pot_balance",
+        name="Balance",
+        value=lambda data: data["balance"] / 100,
+        device_class=SensorDeviceClass.MONETARY,
+        native_unit_of_measurement="£",
+        suggested_display_precision=2,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Defer sensor setup to the shared sensor module."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][CONF_COORDINATOR]
+
+    entities = []
+    for index, account in enumerate(hass.data[DOMAIN][config_entry.entry_id][ACCOUNTS]):
+        for entity_description in ACC_SENSORS:
+            entities.append(
+                MonzoSensor(
+                    coordinator,
+                    entity_description,
+                    index,
+                    account["name"],
+                    lambda x: x["accounts"],
+                )
+            )
+
+    for index, _pot in enumerate(hass.data[DOMAIN][config_entry.entry_id][POTS]):
+        for entity_description in POT_SENSORS:
+            entities.append(
+                MonzoSensor(
+                    coordinator, entity_description, index, "Pot", lambda x: x["pots"]
+                )
+            )
+
+    async_add_entities(entities)
+
+
+class MonzoSensor(MonzoBaseEntity, SensorEntity):
+    """Represents a Monzo sensor."""
+
+    entity_description: MonzoSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        entity_description: MonzoSensorEntityDescription,
+        index: int,
+        device_model: str,
+        data_accessor: Callable[
+            [dict[str, list[dict[str, Any]]]], list[dict[str, Any]]
+        ],
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator, index, device_model, data_accessor)
+        self.entity_description = entity_description
+        self._attr_name = f"{self.data['name']} {entity_description.name}"
```

Comment:
Please use `_attr_has_entity_name = True` instead in your `MonzoBaseEntity` and put your names in the `strings.json`

Reply:
The names of the account type are being provided by the external API so that new/unexpected account types are handled or can be added easily, while pot names are created by the user in the Monzo app. Unless I'm mistaken, that makes it unsuitable for strings.json

Reply:
Fair enough.  `_attr_has_entity_name = True` is still a requirement (https://developers.home-assistant.io/docs/core/entity/#has_entity_name-true-mandatory-for-new-integrations), but leave your names where they are then

Reply:
Actually, reading that you were right in the first place. The device name comes from the api but the entity names are constant, then HA can produce the friendly name. I'll fix it now, thanks!